[WebRTC] vp9_encoder.c: error: arithmetic between floating-point type 'double' and enumeration type 'vpx_bit_depth_t' (aka 'enum vpx_bit_depth')

ddkilzer opened the pull request:
#### abf1ad2d935621795408aaa3a0bbb40947c7db31
<pre>
[WebRTC] vp9_encoder.c: error: arithmetic between floating-point type &apos;double&apos; and enumeration type &apos;vpx_bit_depth_t&apos; (aka &apos;enum vpx_bit_depth&apos;)
<a href="https://bugs.webkit.org/show_bug.cgi?id=267148">https://bugs.webkit.org/show_bug.cgi?id=267148</a>
&lt;<a href="https://rdar.apple.com/120384890">rdar://120384890</a>&gt;

Reviewed by Darin Adler.

Use a C-style cast to convert an enum to a double value.  Make sure to
define the WEBRTC_WEBKIT_BUILD macro consistently across all libwebrtc
sub-projects.

* Source/ThirdParty/libwebrtc/Configurations/Base-libwebrtc.xcconfig:
(GCC_PREPROCESSOR_DEFINITIONS):
- Remove two definitions of WEBRTC_WEBKIT_BUILD.
* Source/ThirdParty/libwebrtc/Configurations/Base.xcconfig:
(GCC_PREPROCESSOR_DEFINITIONS): Add WEBRTC_WEBKIT_BUILD.
* Source/ThirdParty/libwebrtc/Configurations/BaseTarget-libaom.xcconfig:
(GCC_PREPROCESSOR_DEFINITIONS):
- Remove WEBRTC_WEBKIT_BUILD.
* Source/ThirdParty/libwebrtc/Configurations/libwebm.xcconfig:
(GCC_PREPROCESSOR_DEFINITIONS): Remove.
- Now defined in Base.xcconfig.
* Source/ThirdParty/libwebrtc/Configurations/opus.xcconfig:
(GCC_PREPROCESSOR_DEFINITIONS):
- Remove WEBRTC_WEBKIT_BUILD.

* Source/ThirdParty/libwebrtc/Source/third_party/libvpx/source/libvpx/vp9/encoder/vp9_encoder.c:
(update_level_info):
- Cast cm-&gt;bit_depth from an enum to a double value to fix the
  warning.

Canonical link: <a href="https://commits.webkit.org/272771@main">https://commits.webkit.org/272771@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/570f05f0a8c45b78b154895ccbd7a4fe2b5442d5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32899 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11657 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34776 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/35523 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29724 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13997 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8833 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/29166 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33355 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9822 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/29381 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/8558 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/8703 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/29337 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36855 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29879 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29746 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/34838 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8834 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6786 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32693 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10525 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9439 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4248 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9458 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/ThirdParty/libwebrtc/Source/third_party/libvpx/source/libvpx/vp9/encoder/vp9_encoder.c:
```diff
@@ -6219,9 +6219,15 @@ static void update_level_info(VP9_COMP *cpi, size_t *size, int arf_src_index) {
   }
 
   // update compression_ratio
+#ifdef WEBRTC_WEBKIT_BUILD
```
Not sure why this is in an `#ifdef`. Not harmful on any platform as far as I can tell.

ddkilzer replied:
This is the style/pattern that's been used in `Source/ThirdParty/libwebrtc` for a while now.

Cc @youennf

ddkilzer replied:
Probably because we don't want to add `<wtf/Platform.h>` dependencies to third-party sources.

darinadler replied:
That argument is backwards! I am suggesting an unconditional change rather than one with an `#if`

youennf replied:
WEBRTC_WEBKIT_BUILD was historically used to disable some code in libwebrtc we did not like.
This is now used to track all WebKit specific changes done in our libwebrtc fork.
This helps when resyncing libwebrtc code from upstream.

Commit:
[WebRTC] vp9_encoder.c: error: arithmetic between floating-point type 'double' and enumeration type 'vpx_bit_depth_t' (aka 'enum vpx_bit_depth')
https://bugs.webkit.org/show_bug.cgi?id=267148
<rdar://120384890>

Reviewed by Darin Adler.

Use a C-style cast to convert an enum to a double value.  Make sure to
define the WEBRTC_WEBKIT_BUILD macro consistently across all libwebrtc
sub-projects.

* Source/ThirdParty/libwebrtc/Configurations/Base-libwebrtc.xcconfig:
(GCC_PREPROCESSOR_DEFINITIONS):
- Remove two definitions of WEBRTC_WEBKIT_BUILD.
* Source/ThirdParty/libwebrtc/Configurations/Base.xcconfig:
(GCC_PREPROCESSOR_DEFINITIONS): Add WEBRTC_WEBKIT_BUILD.
* Source/ThirdParty/libwebrtc/Configurations/BaseTarget-libaom.xcconfig:
(GCC_PREPROCESSOR_DEFINITIONS):
- Remove WEBRTC_WEBKIT_BUILD.
* Source/ThirdParty/libwebrtc/Configurations/libwebm.xcconfig:
(GCC_PREPROCESSOR_DEFINITIONS): Remove.
- Now defined in Base.xcconfig.
* Source/ThirdParty/libwebrtc/Configurations/opus.xcconfig:
(GCC_PREPROCESSOR_DEFINITIONS):
- Remove WEBRTC_WEBKIT_BUILD.

* Source/ThirdParty/libwebrtc/Source/third_party/libvpx/source/libvpx/vp9/encoder/vp9_encoder.c:
(update_level_info):
- Cast cm->bit_depth from an enum to a double value to fix the
  warning.

Canonical link: https://commits.webkit.org/272771@main
